### PR TITLE
fixes #18773 - Enables content overrides bulk host

### PR DESCRIPTION
**Pull request by parthaa**
This commit includes api changes to the enable setting content overrides for hosts in bulk.

### CODE REVIEW

**Comment by parthaa** [comment]
blocked by https://github.com/Katello/katello/pull/6667

**Comment by parthaa** [comment]
Tested with
```bash
$ hammer  host subscription product-content --host-id=2
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------
ID   | NAME                                                     | TYPE | URL | GPG KEY | LABEL                          | ENABLED? | OVERRIDE             
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------
168  | Red Hat Enterprise Linux 6 Server (RPMs)                 |      |     |         | rhel-6-server-rpms             | yes      | 
5117 | Red Hat Virtualization 4 Agents for RHEL 6 Server (RPMs) |      |     |         | rhel-6-server-rhv-4-agent-rpms | no       |                      
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------

# Add Enable override to "rhel-6-server-rpms"

$ http PUT :3000/api/v2/hosts/bulk/content_overrides \
    organization_id=12 \
    included:='{"ids": [2]}' \
    content_overrides:='[{"content_label":"rhel-6-server-rpms", "name":"enabled", "value": "true"}]'

.....

# Check the result
$ hammer  host subscription product-content --host-id=2
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------
ID   | NAME                                                     | TYPE | URL | GPG KEY | LABEL                          | ENABLED? | OVERRIDE             
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------
168  | Red Hat Enterprise Linux 6 Server (RPMs)                 |      |     |         | rhel-6-server-rpms             | yes      | enabled:1, priority:1
5117 | Red Hat Virtualization 4 Agents for RHEL 6 Server (RPMs) |      |     |         | rhel-6-server-rhv-4-agent-rpms | no       |                      
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------


# Remove Enable override for "rhel-6-server-rpms"
$ http PUT :3000/api/v2/hosts/bulk/content_overrides \
    organization_id=12 \
    included:='{"ids": [2]}' \
    content_overrides:='[{"content_label":"rhel-6-server-rpms", "name":"enabled", "remove": true}]'
.....

# Check the result
$ hammer  host subscription product-content --host-id=2
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------
ID   | NAME                                                     | TYPE | URL | GPG KEY | LABEL                          | ENABLED? | OVERRIDE             
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------
168  | Red Hat Enterprise Linux 6 Server (RPMs)                 |      |     |         | rhel-6-server-rpms             | yes      | enabled:1, priority:1
5117 | Red Hat Virtualization 4 Agents for RHEL 6 Server (RPMs) |      |     |         | rhel-6-server-rhv-4-agent-rpms | no       |                      
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------


# Add multiple overiddes to the host for "rhel-6-server-rpms" and a non existent label "rhel-8-server-rpms"
# Basically overrides should only be applied to hosts that have that content label in the bulk case.
#
$ http PUT :3000/api/v2/hosts/bulk/content_overrides \
    organization_id=12 \
    included:='{"ids": [2]}' \
    content_overrides:='[{"content_label":"rhel-6-server-rpms", "name":"enabled", "value": "true"}, {"content_label":"rhel-6-server-rpms", "name":"priority", "value": "1"}, {"content_label":"rhel-8-server-rpms", "name":"priority", "value": "1"}]'

....

# Check the result
$ hammer  host subscription product-content --host-id=2
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------
ID   | NAME                                                     | TYPE | URL | GPG KEY | LABEL                          | ENABLED? | OVERRIDE             
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------
168  | Red Hat Enterprise Linux 6 Server (RPMs)                 |      |     |         | rhel-6-server-rpms             | yes      | enabled:1, priority:1
5117 | Red Hat Virtualization 4 Agents for RHEL 6 Server (RPMs) |      |     |         | rhel-6-server-rhv-4-agent-rpms | no       |                      
-----|----------------------------------------------------------|------|-----|---------|--------------------------------|----------|----------------------

```

**Comment by parthaa** [comment]
Curl equivalent of my httpie commands
```
curl -X PUT -k -u admin:changeme -H "Content-Type: application/json" -d '{"organization_id":12, "included": {"ids": [2]}, "content_overrides":[{"content_label":"rhel-6-server-rpms", "name":"enabled", "value": "true"}]}' localhost:3000/api/v2/hosts/bulk/content_overrides

```

File: test/controllers/api/v2/host_subscriptions_controller_test.rb [hunk 1]
```diff
@@ -220,7 +222,7 @@ def test_content_override_bulk
     def test_content_override_accepts_string_values
       content_label = "some-content"
       value = "1"
-      assert_sync_task(::Actions::Katello::Host::UpdateContentOverrides) do |host, overrides|
+      assert_sync_task(::Actions::Katello::Host::UpdateContentOverrides) do |host, overrides, _|
```
Comment (thomasmckay) [inline test/controllers/api/v2/host_subscriptions_controller_test.rb]: Is the underscore var name common? I've not seen that before.

Comment (parthaa) [inline test/controllers/api/v2/host_subscriptions_controller_test.rb]: yeah '_' is the convention if you are not planning to use that variable. Else you will get an error from rubcop complaining "unused" variable

**Comment by thomasmckay** [comment]
ack

**Pull request merged**